pkg/admission/handlers/sts.go

```diff
@@ -0,0 +1,66 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/kanopy-platform/hedgetrimmer/internal/admission"
+	"github.com/kanopy-platform/hedgetrimmer/internal/limitrange"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	kadmission "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type STSHandler struct {
+	DefaultDecoderInjector
+	ptm admission.PodTemplateSpecMutator
+}
+
+func NewSTSHandler(ptm admission.PodTemplateSpecMutator) *STSHandler {
+	return &STSHandler{ptm: ptm}
+}
+
+func (sts *STSHandler) Kind() string { return "StatefulSet" }
+
+func (sts *STSHandler) Handle(ctx context.Context, req kadmission.Request) kadmission.Response {
+	log := log.FromContext(ctx)
+
+	lrConfig := ctx.Value(admission.LimitRangeContextTypeMemory).(*limitrange.Config)
```

Comment:
This check should use the non-panic version.

```
	lrConfig, ok := ctx.Value(admission.LimitRangeContextTypeMemory).(*limitrange.Config)
	if !ok {
	}
```

There's no need to check `lrConfig == nil` I think, this [playground](https://go.dev/play/p/Jb0LJspcjWm) shows that nil will also trigger !ok